subsys/net/ip/net_context.c

```diff
@@ -2520,6 +2520,34 @@ static int recv_raw(struct net_context *context,
 	return ret;
 }
 
+int net_context_recvfrom(struct net_context *context,
+		     net_context_recv_cb_t cb,
+		     k_timeout_t timeout,
+		     void *user_data)
+{
+	int ret = -EIO;
+	NET_ASSERT(context);
+
+	if (!net_context_is_used(context)) {
+		return -EBADF;
+	}
+
+	k_mutex_lock(&context->lock, K_FOREVER);
+
+	if (IS_ENABLED(CONFIG_NET_OFFLOAD) &&
```

Comment:
We can't introduce net_context API functions with offloaded-only implementation.